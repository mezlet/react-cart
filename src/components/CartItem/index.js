import React from 'react';
import PropTypes from 'prop-types';
import AddItemButton from '../../components/AddItemButton';

const CartItem = ({ item, addItem, symbol }) => {
	return (
		<div className='d-flex justify-content-center '>
            <div className='my-3 '>
                <img src={item.image_url} alt='' width="125" height="100" className='m-5' />
                <div  className='text-center'>
                    <span className='d-block'>{item.title}</span>
                    <span className='d-block px-4 mx-2'>
                        From {symbol}
                        {item.price}
                    </span>
                    <AddItemButton  item={item} click={() => addItem(item)} />
                </div>

            </div>
		</div>
	);
};

CartItem.propTypes = {
	item: PropTypes.object.isRequired,
	symbol: PropTypes.string.isRequired,
	addItem: PropTypes.func.isRequired
}
export default CartItem;
