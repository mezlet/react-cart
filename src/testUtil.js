import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
    return checkPropTypes(component.propTypes, expectedProps, 'props', component.name)
}