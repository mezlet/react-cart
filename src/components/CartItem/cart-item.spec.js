import React from 'react';
import { shallow } from 'enzyme';
import CartItem from './index';
import { checkProps} from '../../testUtil';

const props = {
    item: {}, 
    addItem: jest.fn(), 
    symbol: ''
}

const setUp = (props = {}) => {
    const component = shallow(<CartItem {...props} />);
    return component;
}

describe('CartItem', () => {
    let component;
    beforeEach(() => {
        component = setUp(props);
    })
    it('should render without errors', () => {
        expect(component).toMatchSnapshot(); 
    });
    it('should render cart div', () => {
        const wrapper = component.find(`[data-test='cart-div']`);
        expect(wrapper.length).toBe(1); 
    })
    it('Checking PropTypes should not throw error', () => {
        const propErr = checkProps(CartItem, props);
        expect(propErr).toBeUndefined();
    })
})