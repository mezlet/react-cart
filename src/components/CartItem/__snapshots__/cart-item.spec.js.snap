// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem should render without errors 1`] = `ShallowWrapper {}`;
